subs/pantry/src/Pantry.hs

```diff
@@ -626,7 +628,32 @@ loadFromURL
   => Text -- ^ url
   -> Maybe BlobKey
   -> RIO env ByteString
-loadFromURL = undefined
+loadFromURL url Nothing = do
+  loadWithCheck url $ \_ -> return ()
+loadFromURL url (Just bkey@(BlobKey sha size)) = do
+  mcached <- withStorage $ loadURLBlob url bkey
```

Comment:
This isn't what I had in mind. It should be using the URL-based cache for the `loadFromURL url Nothing` case, _not_ this case. For this case: it should only ever load by SHA.

Reply:
Please check out updated version